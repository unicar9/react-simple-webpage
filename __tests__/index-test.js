import React from 'react';
import { NewsList, BigCards, MidCards, SmallCards } from '../app/index.js';
import renderer from 'react-test-renderer';

/* Invariant Violation: _registerComponent(...): Target container is not a DOM element.
this error occured when I export obove components inside the same file that is calling ReactDOM.render.
jest.mock('react-dom') is to avoid the conflict*/
jest.mock('react-dom');


describe('whole webpage renders correctly', () => {
  it('BigCards renders correctly', () => {
    const component = renderer.create(<BigCards />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('MidCards renders correctly', () => {
    const component = renderer.create(<MidCards />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('SmallCards renders correctly', () => {
    const component = renderer.create(<SmallCards />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('NewsList renders correctly', () => {
    const component = renderer.create(<NewsList />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
