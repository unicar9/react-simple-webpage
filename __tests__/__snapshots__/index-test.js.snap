// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`whole webpage renders correctly BigCards renders correctly 1`] = `
<div>
  <img
    src="http://lorempixel.com/400/200/sports/undefined"
  />
  <p />
</div>
`;

exports[`whole webpage renders correctly MidCards renders correctly 1`] = `
<div>
  <img
    src="http://lorempixel.com/200/100/food/undefined"
  />
  <p />
</div>
`;

exports[`whole webpage renders correctly NewsList renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="big"
  >
    <div>
      <img
        src="http://lorempixel.com/400/200/sports/1"
      />
      <p>
        No apologies James Taylor no right answer my scarf we are. Never getting back together candles Tree Paine all too.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/400/200/sports/2"
      />
      <p>
        No apologies James Taylor no right answer my scarf we are. Never getting back together candles Tree Paine all too.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/400/200/sports/3"
      />
      <p>
        No apologies James Taylor no right answer my scarf we are. Never getting back together candles Tree Paine all too.
      </p>
    </div>
  </div>
  <div
    className="mid"
  >
    <div>
      <img
        src="http://lorempixel.com/200/100/food/0"
      />
      <p>
        Sad all too well presumptuous no apologies no apologies break and.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/200/100/food/1"
      />
      <p>
        Sad all too well presumptuous no apologies no apologies break and.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/200/100/food/2"
      />
      <p>
        Sad all too well presumptuous no apologies no apologies break and.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/200/100/food/3"
      />
      <p>
        Sad all too well presumptuous no apologies no apologies break and.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/200/100/food/4"
      />
      <p>
        Sad all too well presumptuous no apologies no apologies break and.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/200/100/food/5"
      />
      <p>
        Sad all too well presumptuous no apologies no apologies break and.
      </p>
    </div>
    <div>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      </p>
    </div>
    <div>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      </p>
    </div>
    <div>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/200/100/food/5"
      />
      <p>
        Sad all too well presumptuous no apologies no apologies break and.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/200/100/food/6"
      />
      <p>
        Sad all too well presumptuous no apologies no apologies break and.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/200/100/food/7"
      />
      <p>
        Sad all too well presumptuous no apologies no apologies break and.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/200/100/food/8"
      />
      <p>
        Sad all too well presumptuous no apologies no apologies break and.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/200/100/food/9"
      />
      <p>
        Sad all too well presumptuous no apologies no apologies break and.
      </p>
    </div>
    <div>
      <img
        src="http://lorempixel.com/200/100/food/10"
      />
      <p>
        Sad all too well presumptuous no apologies no apologies break and.
      </p>
    </div>
    <div>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      </p>
    </div>
    <div>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      </p>
    </div>
    <div>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      </p>
    </div>
  </div>
</div>
`;

exports[`whole webpage renders correctly SmallCards renders correctly 1`] = `
<div>
  <p />
</div>
`;
